docs/rules/expiring-todo-comments.md

```diff
@@ -0,0 +1,108 @@
+# Add conditions to TODO comments to make them error
+
+Makes it possible to pass arguments to TODO and FIXME comments to trigger errors.
+
+This rule also implements [`eslint/no-warning-comments`](https://eslint.org/docs/rules/no-warning-comments) behavior so you should disable that in favor of using this.
+
+For projects with legacy branches, such as long term supported older versions, there might be situations where you fix TODOs in the master branch but on the other branches you leave it missing. Choose carefully which projects you would want this feature.
+
+Supported conditions:
+
+- `[YYYY-MM-DD]` to define a due date.
+- `[>1]` or `[>=2]` to expire at some version (from package.json). No whitespace.
+- `[+package]` or `[-package]` to expire when you add/remove a package.
+- `[package@>1]` or `[package@>=2]` to expire when a package hits some version. No whitespace.
+- `[engines node>8]` or `[engines node>=8]` to expire when package bump supported engines. Don't use spaces between the comparison.
```

Comment:
Having the word `or` here makes it sound like only those two semver notation are supported.

Reply:
Actually it is.

On my first draft I used also '=', '<' and '<='. Is this desirable?

Reply:
Nah, we can keep it simple. I would just make it clear with a sentence, that while it looks like we support all of semver (which I just assumed when seeing semver syntax), we only support ...

Reply:
Done